src/openfermion/measurements/_qubit_partitioning.py

```diff
@@ -0,0 +1,167 @@
+    such that all pairs of qubits are split in at least one partition,
+    This follows a variation on ArXiv:1908.0562 - instead of
+    explicitly partitioning the list based on the binary indices of
+    the qubits, we repeatedly divide the list in two and then
+    zip it back together.
+
+    Args:
+        qubit_list(list): list of qubits to be partitioned
+        num_iterations(int or None): number of iterations to perform.
+            If None, will be set to ceil(log2(len(qubit_list)))
+
+    Returns:
+        partition(iterator of lists of lists): the required partitioning
+    '''
+
+    # Some edge cases
+    if num_iterations is not None and num_iterations == 0:
+        return
+    num_qubits = len(qubit_list)
+    if num_qubits < 2:
+        raise ValueError('Need at least 2 qubits to partition')
+    if num_qubits == 2:
+        yield [[qubit_list[0]], [qubit_list[1]]]
+        return
+
+    num_iterations = num_iterations or\
+        int(numpy.ceil(numpy.log2(num_qubits)))
+
+    # Calculate the point where we need to split the list each time.
+    half_point = int(numpy.ceil(num_qubits/2))
+
+    # Repeat the division and zip steps as many times
+    # as required.
+    for j in range(num_iterations):
+        # Divide the qubit list in two and return it
+        partition = [qubit_list[:half_point],
+                     qubit_list[half_point:]]
+        yield partition
+        # Zip the partition together to remake the qubit list.
+        qubit_list = list(chain(*zip_longest(partition[0], partition[1])))
+        # If len(qubit_list) is odd, the end of the list will be 'None'
+        # which we delete.
+        if qubit_list[-1] is None:
+            del qubit_list[-1]
+        
+
+
+def partition_iterator(qubit_list, partition_size, num_iterations=None):
+    '''Generator for a list of k-partitions of N qubits such that
+    all sets of k qubits are perfectly split in at least one
+    partition, following ArXiv:1908.05628
+
+    Args:
+        qubit_list(list): list of qubits to be partitioned
+        partition_size(int): the number of partitions.
+        num_iterations(int or None): the number of iterations in the
+            outer iterator. If None, set to ceil(log2(len(qubit_list)))
+
+    Returns:
+        partition(iterator of lists of lists): the required partitioning
+    '''
+
+    # Some edge cases
+    if num_iterations is not None and num_iterations == 0:
+        return
+    if partition_size == 1:
+        yield [qubit_list]
+        return
+    elif partition_size == 2:
+        for p in binary_partition_iterator(qubit_list, num_iterations):
+            yield p
+        return
+    num_qubits = len(qubit_list)
+    if partition_size == num_qubits:
+        yield [[q] for q in qubit_list]
+        return
+    elif partition_size > num_qubits:
+        raise ValueError('I cant k-partition less than k qubits')
+
+
+    num_iterations = num_iterations or\
+        int(numpy.ceil(numpy.log2(num_qubits)))
```

Comment:
Checking for the None case looks better than doing a line continuation:
```
if num_iterations is None:
    num_iterations = ...
```

Reply:
Changed